### DISCUSSION TITLE
Generic Entity Object vs. TypedEntities

### DISCUSSION THREAD

**Discussion by mdx-mrph** [jordimontana82/fake-xrm-easy#577] [Q&A] [answered]
Hi everyone,

I'm very new to C# and Plugin development, and I'm trying to run some tests to understand what is returned from queries. 

What I'm wondering is whether or not Typed Entities (Account, Contact, Incident, etc) are still supported? I've been trying to figure out if I'm missing a reference somewhere, or people are creating custom classes to use these objects. I only seem to be able to access the generic Entity object without receiving a reference error. I'm sure this is a very silly question, but would someone be able to help me?

**Comment by jordimontana82** [comment, marked as answer]
Hi,
These early bound classes are typically generated with crmsvcutil, which is part of the tools Nuget pacakge download or similar tools like EarlyBoundGenerator or AlbanianEarlyBound generators in XrmToolBox.

Few links:

https://docs.microsoft.com/en-US/dynamics365/customerengagement/on-premises/developer/org-service/create-early-bound-entity-classes-code-generation-tool?view=op-9-1

https://www.xrmtoolbox.com/plugins/DLaB.Xrm.EarlyBoundGenerator/ -> if you have questions @daryllabar  is your man

https://www.xrmtoolbox.com/plugins/AlbanianXrm.EarlyBound/  -> ping @BetimBeja  for this one.

Ideally is way better to use strongly typed entity objects because type inference works better, generally speaking.

**Reply by mdx-mrph** [reply]
Hey Jordi,
I can't believe the promptness of this response.... thank you so much for the answer. Going to investigate this all now.
